### ISSUE TITLE
support parsing scientific notation doubles

### ISSUE THREAD

**Issue by filippog** [armon/statsite#123]
hi,
looks like `str2double` doesn't support parsing numbers in scientific notation, e.g. diamond would send metrics like this `diskspace._var.byte_used:7.72529e+11|g` and gets parsed to `7.72529`

**Comment by armon** [comment]
Is this a format statsd supports? I didn't realize this was valid input

**Comment by filippog** [comment]
I think so, yeah:

```
 gauges: 
   { 'diskspace._var.byte_used': 772529000000,
```
also
```
   { 'diskspace.ar.bytesed': 1.72529e-9,
```

both `1.23e-4` and `1.23e-04` seem recognized and case doesn't matter like `strtod`

**Comment by armon** [comment]
Interesting. I found that strtod introduced a large performance penalty in the parser, so maybe I will just add an unlikely path to detect the `e` and jump out to strtod.

**Comment by filippog** [comment]
sth like in #124 ?

related to this, what would be a good way to test static functions with check? `#include 'conn_handler.c'` comes to mind but then conditionally include `conn_handler.h` inside to avoid the double definition?

**Comment by theatrus** [comment]
Curious under what circumstances you had worse performance with strtod - all of my testing has shown a 15% penalty to str2double.

**Comment by armon** [comment]
Fixed by #124. Thanks!
